### ISSUE TITLE
Cancel jobs on fast_finish?

### ISSUE THREAD

**Issue by svenfuchs** [travis-ci/travis-ci#2062]
We have an option to finish the build early when the option `fast_finish` was given. But this (intentionally) does not cancel the rest of the build jobs, which continue to run.

A customer would like to cancel all jobs once one job has failed.

**Comment by joshk** [comment]
Closing this as it has become stale.

**Comment by mattkasa** [comment]
Stale?  It still seems like a relevant feature request, unless there's another way to do it that you can recommend?

**Comment by joshk** [comment]
Hi @mattkasa 

I'm sorry, but as we are not likely to implement this change in the next 3-6 months, I am going to close it for the time being.

**Comment by auroraeosrose** [comment]
You really should have a "won't fix" label if you're going to close feature requests like this

**Comment by cdown** [comment]
Agreed, it doesn't make sense to close meaningful tasks just because they are not on your roadmap. Milestone them, or label them, but don't throw away your best method of development: user feedback.

**Comment by mtola** [comment]
so what's new for that because, yes, fast_finish: true don't work :-(

**Comment by Arcanemagus** [comment]
I would imagine this would be a higher priority feature request as it would allow useless builds to be cancelled early on, saving resources from being spent on them.

**Comment by kgryte** [comment]
Any update on this? This is also a feature that I would like implemented.

**Comment by joshk** [comment]
I wanted to let everyone know that I have reviewed this item and have added it to our Roadmap. We are about to release a new feature, which this would fit in as a nice complimentary. I expect it to fit in Q1 of 2017.

**Comment by WaleedAshraf** [comment]
Expecting this to release withi  few weeks.
Yaaaa! Much awaited.

**Comment by BanzaiMan** [comment]
Sorry about the delay and confusion. There was a change in planning, and this has been pushed back.

**Comment by native-api** [comment]
Well, if you have so much free buildbot machine time you can tolerate it being wasted with impunity...

**Comment by native-api** [comment]
At this moment, I see Mac jobs spending literally days in the queue, so I seriously doubt this is the case.

**Comment by monperrus** [comment]
I would be much interested in this feature.

**Comment by ringerc** [comment]
Following. I'd love to have sentinel builds.

**Comment by birdbrained** [comment]
This feature is something that would be very beneficial to have, for example to conserve build resources one could cancel tests that run in parallel if linting fails.

**Comment by jayvdb** [comment]
I suspect many users needs can be met by using job stages, using the stage ordering, as you can put a sentinel job (love that name) in a 'sentinel' stage, and put the 'test' stage after it.
e.g. https://github.com/coala/coala-bears/pull/2541 and https://travis-ci.org/coala/coala/builds/393939619

**Comment by boxrick** [comment]
I am just in the process of looking at this, it feels like a massive waste of resources to keep parallel jobs running if one of them fails. I want to fail fast not sit around waiting for old cruft to finish even if the build has failed. Anyone have any sensible workarounds for this issue?

**Comment by JoshuaSBrown** [comment]
Is there a new timeline for the implementation of this feature?

**Comment by rodrigodgm** [comment]
What about this feature, I don't think is useful but NEEDED

**Comment by mmcgahan** [comment]
Would love to see this implemented, but a sort of hacky workaround is to force-cancel the build using the [build cancel action in the Travis API](https://developer.travis-ci.com/resource/build#cancel) in every job's `after_failure` - it won't set the overall build status to 'Failed', which would be preferable/accurate, but it can at least fail fast and stop other long-running parallel jobs when one job fails.

1. Get an API key from https://travis-ci.com/account/preferences
2. Set it as a 'secret' env var in your build (e.g. in the web GUI https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings) - I called it `TRAVIS_API_TOKEN`
3. In each of your jobs (or just the ones that should cause a fast fail):
    ```yaml
    jobs:
      include:
        - name: 'This will cancel the build when it fails'
          script: 'do whatever you want here'
          after_failure:
            - 'curl -X POST -H "Travis-Api-Version: 3" -H "Authorization: token $TRAVIS_API_TOKEN" https://api.travis-ci.com/build/$TRAVIS_BUILD_ID/cancel'
    ```

I haven't tested this much for edge case behavior, but hopefully it can help in some applications

**Comment by tong-luo1** [comment]
Good morning,
We'd also really like to see this feature being implemented soon if possible, since when any job which is allowed to fail didn't start due to server availability issue, the build result would not become available even though all the required jobs finish running.  This is occasionally blocking our PR merge.
Thank you!
